ndcube/tests/test_ndcube.py

```diff
@@ -1179,73 +1152,170 @@ def cube(request):
                                                             uncertainty=StdDevUncertainty(np.ones((10, 12)) * 0.1))
                             ),
                         ],
-                        indirect=["cube"])
-def test_cube_add_one_unit(cube, value):
-    #print(f"cube: {cube}")
-    assert isinstance(cube, NDCube)
+                        indirect=("ndc",))
+def test_cube_add_one_unit(ndc, value):
+    assert isinstance(ndc, NDCube)
     with pytest.raises(TypeError, match="Adding objects requires both have a unit or neither has a unit."):
-        cube + value
+        ndc + value
+
 
-# The case when both NDData and NDCube have uncertainty. No mask is involved.
-# Both NDData NDCube have a unit.
+# Both NDData and NDCube have unit and uncertainty. No mask is involved.
 # Test different scenarios when units are equivalent and when they are not. TODO (bc somewhere is checking the units are the same)
-# 1, user input, expected output
-# 2, check the codebase itself
-@pytest.mark.parametrize('value', [
-    NDData(np.ones((10, 12)), # pass in the values to be tested as a set of ones.
-           wcs=None,
-           unit=u.ct,
-           uncertainty=StdDevUncertainty(np.ones((10, 12))*0.1, unit=u.ct)),
-])
-def test_cube_add_both_unit(ndcube_2d_with_unit_uncertainty, value):
-    new_cube = ndcube_2d_with_unit_uncertainty + value # perform the addition
+# what is an equivalent unit in astropy for count (ct)?
+@pytest.mark.parametrize(("ndc", "value"),
+                        [
+                            ("ndcube_2d_unit_unc", NDData(np.ones((10, 12)), # pass in the values to be tested as a set of ones.
+                                                          wcs=None,
+                                                          unit=u.ct,
+                                                          uncertainty=StdDevUncertainty(np.ones((10, 12))*0.1, unit=u.ct))
+                            ),
+                        ],
+                        indirect=("ndc",))
+def test_cube_add_cube_unit_unc_nddata_unit_unc(ndc, value):
+    new_cube = ndc + value # perform the addition
     # Check uncertainty propagation
-    expected_uncertainty = ndcube_2d_with_unit_uncertainty.uncertainty.propagate(
+    expected_uncertainty = ndc.uncertainty.propagate(
                             operation=np.add,
                             other_nddata=value,
                             result_data=new_cube.data*new_cube.unit,
```

Comment:
This test is a bit circular.  You shouldn't use outputs of the test to form expected values.  Otherwise the test may pass when it shouldn't.
```suggestion
    expected_unit = u.ct
    expected_data = (ndc.data * ndc.unit).to_value(expected_unit) + (value.data * value.unit).to_value(expected_unit)
    expected_uncertainty = ndc.uncertainty.propagate(
                            operation=np.add,
                            other_nddata=value,
                            result_data=expected_data*expected_unit,
```

Reply:
Hi @DanRyanIrish , I checked the code for the assert_cubes_equal method. I am unsure whether it checks the values, type and units of the uncertainty attributes as well. 
It looks like it only checks whether the shapes of uncertainties are the same.

Reply:
@DanRyanIrish Or maybe I can 
either: use the assert_cube_equal method together with a few more lines that checks the values, type and units of the uncertainty attributes, 
or: adding a few more lines into the assert_cube_equal method?

Reply:
I see what you mean, thank you for this suggestion! I will implement this.